### PR TITLE
Buybutton

### PR DESCRIPTION
**Pull request by ak2253**

### CODE REVIEW

File: server/models/store.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 class Store(db.Model):
     id = db.Column(db.Integer, primary_key=True)
     name = db.Column(db.String(255))
+    item_type = db.Column(db.Integer)
```
Comment (dvkm) [inline server/models/store.py]: id will be used as item_type. Not needed.

File: server/routes/store.py [hunk 2]
```diff
@@ -1,8 +1,49 @@
...
+    transaction = Transaction(
+        user_id=user_id,
+        ticket_amount=item_info.price,
+        activity="ticketstore",
+    )
+    db.session.add(transaction)
+    item_query = Item.query.filter_by(user_id=user_id, item_type=item_type).first()
+    if item_query is None:
+        item = Item(
+            item_type = item_type,
```
Comment (dvkm) [inline server/routes/store.py:39]: item_type = id

File: server/routes/store.py [hunk 3]
```diff
@@ -18,3 +59,19 @@ def store_list():
         store_items.append(item_obj)
 
     return {"items": store_items}
+
+@store_bp.route("/buy", methods=["POST"])
+def buy_item():
+    try:
+        data = json.loads(request.data)
+        item_type = data["item_type"]
```
Comment (dvkm) [inline server/routes/store.py]: data ["id"]

**Review by dvkm — APPROVED** [review]
Looks good. Just need some change. Feel free to merge after the changes.

File: server/routes/store.py (hunk 2 above)
Comment (ak2253) [reply server/routes/store.py:39]: why id?

Comment (dvkm) [reply server/routes/store.py:39]: id in store is what item_type should've been.

**Pull request merged**